### PR TITLE
Adding some constants due to additions to hts-spec

### PR DESCRIPTION
**Pull request by yfarjoun**

### CODE REVIEW

File: src/main/java/htsjdk/samtools/util/SamConstants.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package htsjdk.samtools.util;
+
+/**
+ * A set of constants defined in the sam-spec (https://github.com/samtools/hts-specs) that need
+ * to be referenced in code.
+ */
+public class SamConstants {
```
Comment (lbergelson) [inline src/main/java/htsjdk/samtools/util/SamConstants.java]: make this final...
```suggestion
 public final class SamConstants {
```

and I'd give it a private constructor as well, but I know that's more controversial...

File: src/main/java/htsjdk/samtools/util/SamConstants.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+    final static String BARCODE_SEQUENCE_DELIMITER = "-";
+    final static String BARCODE_QUALITY_DELIMITER = "~";
```
[the remark this answers is not in the document]
Comment (lbergelson) [reply src/main/java/htsjdk/samtools/util/SamConstants.java]: Ooh, nice catch.  @yfarjoun Explain yourself!

**Comment by yfarjoun** [comment]
can this be merged now?

File: src/main/java/htsjdk/samtools/util/SamConstants.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (yfarjoun) [reply src/main/java/htsjdk/samtools/util/SamConstants.java]: fixed

**Comment by yfarjoun** [comment]
@lbergelson...I took your suggestions.

**Pull request merged**